src/jit/emitxarch.cpp

```diff
@@ -13663,6 +13663,1128 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_25C;
+            break;
+
+        case INS_mov_xmm2i:
+            // movd  reg, xmm
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_2C;
+            break;
+
+        case INS_mov_i2xmm:
+            // movd  xmm, reg
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_1C;
+            break;
+
+        case INS_movq:
+            if (memAccessKind == PERFSCORE_MEMORY_NONE)
+            {
+                // movq   reg, reg
+                result.insThroughput = PERFSCORE_THROUGHPUT_3X;
+                result.insLatency    = PERFSCORE_LATENCY_1C;
+            }
+            else if (memAccessKind == PERFSCORE_MEMORY_READ)
+            {
+                // movq   reg, mem
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+            }
+            else
+            {
+                // movq   mem, reg
+                assert(memAccessKind == PERFSCORE_MEMORY_WRITE);
+                result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            }
+            break;
+
+        case INS_movdqa:
+        case INS_movdqu:
+            if (memAccessKind == PERFSCORE_MEMORY_NONE)
+            {
+                // ins   reg, reg
+                result.insThroughput = PERFSCORE_THROUGHPUT_4X;
+                result.insLatency    = PERFSCORE_LATENCY_ZERO;
+            }
+            else if (memAccessKind == PERFSCORE_MEMORY_READ)
+            {
+                // ins   reg, mem
+                result.insThroughput = PERFSCORE_THROUGHPUT_2X;
+            }
+            else
+            {
+                // ins   mem, reg
+                assert(memAccessKind == PERFSCORE_MEMORY_WRITE);
+                result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            }
+            break;
+
+        case INS_movhps:
+        case INS_movhpd:
+        case INS_movlps:
+        case INS_movlpd:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            if (memAccessKind == PERFSCORE_MEMORY_READ)
+            {
+                result.insLatency = max(PERFSCORE_LATENCY_4C, result.insLatency);
+            }
+            else
+            {
+                assert(memAccessKind == PERFSCORE_MEMORY_WRITE);
+                result.insLatency = max(PERFSCORE_LATENCY_3C, result.insLatency);
+            }
+            break;
+
+        case INS_movhlps:
+        case INS_movlhps:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
+            result.insLatency    = PERFSCORE_LATENCY_1C;
+            break;
+
+        case INS_xorps:
+            // uops.info
+            result.insThroughput = PERFSCORE_THROUGHPUT_3X;
+            result.insLatency    = PERFSCORE_LATENCY_ZERO;
+            break;
+
+        case INS_vzeroupper:
+            result.insThroughput = PERFSCORE_THROUGHPUT_1C;
```

Comment:
I believe this is significantly more expensive on older processors, if that matters.

Skylake in particular was updated to have better behavior with `vzeroupper`

Reply:
I have modeled the Intel Skylake architecture here.
Alternatively we could increase the cost for the instructions that are slower on AMD Ryzen.

Reply:
This is also fast on Ryzen (and is frequently elided in the register renamer). It's slower on older processors from both AMD and Intel (where it is not elided in the register renamer and where transition penalties between VEX and legacy SIMD instructions is greater).

I wonder if Skylake in particular is the best to model, considering its from ~2015 and most of the Azure cloud machines (and looks to be similar for others, like AWS) look to be Haswell and Broadwell Xeon CPUs (although some specific higher-end machines are the Skylake Scalable CPU).